source/vibe/http/server.d

```diff
@@ -707,28 +707,47 @@ final class HTTPServerRequest : HTTPRequest {
 	@property URL fullURL()
 	const {
 		URL url;
-		auto fh = this.headers.get("X-Forwarded-Host", "");
-		if (!fh.empty) {
-			url.schema = this.headers.get("X-Forwarded-Proto", "http");
-			url.host = fh;
+
+		// Set URL host segment.
+		if (auto xfh = this.headers.get("X-Forwarded-Host")) {
+			url.host = xfh;
+		} else if (!this.host.empty) {
+			url.host = this.host;
+		} else if (!m_settings.hostName.empty) {
+			url.host = m_settings.hostName;
 		} else {
-			if (!this.host.empty) url.host = this.host;
-			else if (!m_settings.hostName.empty) url.host = m_settings.hostName;
-			else url.host = m_settings.bindAddresses[0];
+			url.host = m_settings.bindAddresses[0];
+		}
 
-			if (this.tls) {
-				url.schema = "https";
-				if (m_port != 443) url.port = m_port;
-			} else {
-				url.schema = "http";
-				if (m_port != 80) url.port = m_port;
+		// Set URL schema segment.
+		if (auto xfp = this.headers.get("X-Forwarded-Proto")) {
+			url.schema = xfp;
```

Comment:
I'd argue this should set the port to the default port for that schema for cases where X-Forwarded-Port isn't set.

Reply:
Setting no port (defaut of `0`) is considered to be implicitly equal to setting the default port. Are you thinking of something in particular that is broken?